query.go

```diff
@@ -123,3 +123,15 @@ func (q *Query) SetExcludeScheme(mode ExcludeMode) {
 	cmode := C.notmuch_exclude_t(mode)
 	C.notmuch_query_set_omit_excluded(q.toC(), cmode)
 }
+
+// AddTagExclude adds a tag that will be excluded from the query results by default.
+// This exclusion will be ignored if this tag appears explicitly in the query.
+func (q *Query) AddTagExclude(tag string) error {
+	ctag := C.CString(tag)
+	defer C.free(unsafe.Pointer(ctag))
+	status := C.notmuch_query_add_tag_exclude(q.toC(), ctag)
+	if status != C.NOTMUCH_STATUS_IGNORED {
+		return statusErr(status)
+	}
```

Comment:
Why are we silencing the error?

Reply:
Because that's a non fatal error, given if one sets an exclude tag but still provides it in the query itself.

So say as an example the query `tag:spam` is provided but also sets exclude_tags: spam is set. In this case notmuch will yield STATUS_IGNORED, as obviously the direct query has precedence.

I'm reluctant to push that to the caller as a error, because it's simply a warning. I really want to avoid having to do
```go
err := query.SetExcludeTag(tag)
if err != notmuch.ErrIgnored {
    fmt.Println("ignored")
} else if err != nil {
   log.Fatal("Actually an error")
}
```
After all the query is usually user provided, but often you have a config option (in notmuch's case search.exclude_tags) which is always applied on a query.

Reply:
I'd rather not deviate from the C API unless it's either (1) to make it more idiomatic Go, or (2) to make things memory-safe -- especially in ways that won't get caught by the type system. It's common enough to want to handle a given error specially (frequently io.EOF), and I think it's not too bad to have to write:

```go
err := query.SetExcludeTag(tag)
if err != nil && err != notmuch.ErrIgnored {
    log.Fatal(...)
}
```

Reply:
as you wish